contracts/rewards/src/contract/execute.rs

```diff
@@ -44,20 +45,23 @@ where
     fn current_epoch(&self, cur_block_height: u64) -> Result<Epoch, ContractError> {
         let stored_params = self.store.load_params();
         let epoch_duration: u64 = stored_params.params.epoch_duration.into();
-        let epoch = stored_params.last_updated;
-        if cur_block_height < epoch.block_height_started {
-            return Err(ContractError::BlockHeightInPast.into());
-        }
-        if cur_block_height < epoch.block_height_started + epoch_duration {
-            return Ok(epoch);
-        }
-        let epochs_elapsed = (cur_block_height - epoch.block_height_started) / epoch_duration;
-        let epoch_num = epochs_elapsed + epoch.epoch_num;
-        let block_height_started = epoch.block_height_started + epochs_elapsed * epoch_duration;
+        let last_updated_epoch = stored_params.last_updated;
 
-        Ok(Epoch {
-            epoch_num,
-            block_height_started,
+        let epochs_elapsed = if cur_block_height < last_updated_epoch.block_height_started {
+            Err(ContractError::BlockHeightInPast.into())
+        } else {
+            Ok((cur_block_height - last_updated_epoch.block_height_started) / epoch_duration)
+        };
+
+        epochs_elapsed.map(|epochs_elapsed| {
+            match epochs_elapsed {
+                0 => last_updated_epoch,
+                1.. => Epoch {
+                    epoch_num: last_updated_epoch.epoch_num + epochs_elapsed,
+                    block_height_started: last_updated_epoch.block_height_started
+                        + (epochs_elapsed * epoch_duration), // result is strictly less than cur_block_height, so multiplication is safe
+                },
+            }
         })
```

Comment:
```suggestion
        match cur_block_height {
            _ if cur_block_height < last_updated_epoch.block_height_started => Err(ContractError::BlockHeightInPast.into())
            _ => {
                let epochs_elapsed = (cur_block_height - last_updated_epoch.block_height_started) / epoch_duration;

                Ok(Epoch {
                    epoch_num: last_updated_epoch.epoch_num + epochs_elapsed,
                    block_height_started: last_updated_epoch.block_height_started
                        + (epochs_elapsed * epoch_duration), // result is strictly less than cur_block_height, so multiplication is safe
                })
            }
        }
```
There's no need to differentiate epochs_elapsed=0 and epochs_elapsed>=1